README.md

```diff
@@ -16,6 +16,18 @@
   Moreover, Hoa aims at being a bridge between industrial and research worlds.
 </p>
 
+##What's new
+
+1. Support PHP > 8.0
+2. GitHub Action tests
+3. Use 'replace' in composer.json for updating without changes in code
+
+##Another forked packages
+
+1. [hoa-iterator](https://github.com/hiqdev/hoa-iterator)
+2. [hoa-protocol](https://github.com/hiqdev/hoa-protocol)
+3. [hoa-ruler](https://github.com/hiqdev/Hoa-Ruler)
+
```

Comment:
```
## Зачем этот форк

## Какие изменения сделаны в форке

## Как использовать

## Какие ещё пакеты нужно взять форкнутые чтобы было всё ок

## Как версии оргиниального проекта соотносятся к версиям этого пакета


```

Reply:
## Reasons to fork

The Hoa project was archived, and no upgrades or patches are neither provided nor accepted by merge requests.

HOA Packages include some code that is Deprecated for PHP 8.0 and PHP 8.1, but we needed these packages to run on modern PHP versions.

## What's changed in from?

The changes mainly affected the return data type hinting in methods declaration, access to uninitialized properties.

We've also added a pipeline on GitHub Actions to make sure tests are green.

## How to use

We've currently forked the following packages, primarily to make `hoa/ruler` work with PHP 8.1:

| Original package | Forked package |
| -- | -- |
| `hoa/ruler` | `hiqdev/hoa-ruler` |
| `hoa/compiler` | `hiqdev/hoa-compiler` |
| `hoa/protocol` | `hiqdev/hoa-protocol` |
| `hoa/iterator` | `hiqdev/hoa-iterator` |

You can simply replace requirements in `composer.json` from `hoa` packages to the corresponding forked packages: there is no need to change something in the codebase.

If you use someone's package, that requires `hoa` – simply add forks to your project root `composer.json`: we have marked forks as a [replacement](https://getcomposer.org/doc/04-schema.md#replace), so composer will install them instead of the original packages.

## Versions

We've forked from the latest `hoa` package versions and bump own versions starting from `1.0`.

________________